core/build.sh

```diff
@@ -66,11 +67,28 @@ Libraries will be built in 'build/<target>/<configuration>/src'.
   "uninstall       uninstall libraries from current system" \
   "${archtargets[@]}" \
   "--debug,-d                      configuration is 'debug', not 'release'" \
+  "--no-tests      do not configure tests (used by other projects)" \
   "--target-path=opt_target_path   override for build/ target path" \
   "--test=opt_tests,-t             test[s] to run (space separated)"
 
 builder_parse "$@"
 
+#
+# meson forces us to configure tests, including building compilers, even
+# if we don't plan to run them, for example when doing a dependency build
+# in CI
```

Comment:
> including building compilers

If installing a compiler for this is undesirable in CI, maybe you'd prefer to conditionally add the language with `have_somelang = add_languages('somelang', required: tests_mandatory)` and then only configure the tests if the language is available.

Reply:
Thanks for the review!

Sadly, we kinda need to jump through some hoops here anyway: we don't want to build the test dependencies when building, for example, Keyman for Windows, for performance reasons; these dependencies are TypeScript and not built with meson.

I guess the comment in build.sh is a bit misleading in that regard.

The approach we took in the end was to declare a "no tests" option, because it was more explicit, as opposed to previously where we tested for the presence of nodejs, and implicitly disabled certain tests. This means that the default behaviour is to build tests, and fail if something is missing, which is better than silently passing.